source/common/json/json_internal.cc

```diff
@@ -0,0 +1,668 @@
+#include "common/json/json_internal.h"
```

Comment:
qq: this file is all new code, and so needs to be reviewed line-by-line? This is not a copy of of something existing?

Reply:
Sort of, the diff is above (between the legacy and this one). They have some major differences (listed above) that made it difficult to inherit from the same SAX parser.